devito/tools/algorithms.py

```diff
@@ -63,7 +64,10 @@ def toposort(data):
         ordered = set(item for item, dep in mapper.items() if not dep)
         if not ordered:
             break
-        processed = filter_sorted(ordered) + processed
+        try:
+            processed = sorted(ordered, key=attrgetter('name')) + processed
```

Comment:
no filtering?

Reply:
filtering was the issue

I could have two Dimensions called `x` and `x` (yeah, same name), but different objects (even different subtypes)

now it's all handled properly